modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java

```diff
@@ -204,6 +212,7 @@ public void start() {
     @Override
     public void stop() throws Exception {
         metricManager.unregisterSource(metrics);
```

Comment:
1. All the logic in this method is idempotent, so `stopGuard` does not matter.
2. As I understand, `busyLock` won't do anything either. All external input for this module comes through Netty. We can't wait for clients to disconnect in a busy lock. We just drop them by closing the channel.

Reply:
What if we have some pending action along with concurrent stop?

Reply:
There is no such action. `stop` and `start` can't be concurrent (right?), and we don't have any other public methods.

Reply:
If you insist, we can add those things just to be safe and consistent with other components.

Reply:
> There is no such action. stop and start
I'm talking about concurrent stop and e.g. notification(PrimaryReplicaElection) processing. It might be covered with separate ticket in order not to mess given one.

Reply:
Ticket created to address all client-related classes: https://issues.apache.org/jira/browse/IGNITE-20921